dotnet/src/Experimental/Process.Core/ProcessVisualizationExtensions.cs

```diff
@@ -0,0 +1,126 @@
+
+using System;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.SemanticKernel.Process;
+
+/// <summary>
+/// Provides extension methods to visualize a process as a Mermaid diagram.
+/// </summary>
+public static class ProcessVisualizationExtensions
+{
+    /// <summary>
+    /// Generates a Mermaid diagram from a process builder.
+    /// </summary>
+    /// <param name="processBuilder"></param>
+    /// <returns></returns>
+    public static string ToMermaid(this ProcessBuilder processBuilder)
+    {
+        var process = processBuilder.Build();
+        return process.ToMermaid();
+    }
+
+    /// <summary>
+    /// Generates a Mermaid diagram from a kernel process.
+    /// </summary>
+    /// <param name="process"></param>
+    /// <returns></returns>
+    public static string ToMermaid(this KernelProcess process)
+    {
+        StringBuilder sb = new();
+        sb.AppendLine("flowchart LR");
+        //sb.AppendLine("graph LR");
+
+        // Generate the Mermaid flowchart content with indentation
+        string flowchartContent = GenerateMermaidFlowchart(process);
+
+        // Append the formatted content to the main StringBuilder
+        sb.Append(flowchartContent);
+
+        return sb.ToString();
+    }
+
+    /// <summary>
+    /// Generates the Mermaid graph for a given process.
+    /// </summary>
+    /// <param name="process"></param>
+    /// <returns></returns>
+    private static string GenerateMermaidFlowchart(KernelProcess process)
+    {
+        StringBuilder sb = new();
+        string indentation = new(' ', 4);
+
+        // Dictionary to map step IDs to step names
+        var stepNames = process.Steps
+            .Where(step => step.State.Id != null && step.State.Name != null)
+            .ToDictionary(
+                step => step.State.Id!,
+                step => step.State.Name!
+                );
+
+        // Add Start and End nodes with proper Mermaid styling
+        sb.AppendLine($"{indentation}Start[Start]");
+        sb.AppendLine($"{indentation}End[End]");
+
+        // Handle all edges without a predefined "Start"
+        foreach (var kvp in process.Edges)
+        {
+            var stepId = kvp.Key;
+            var edges = kvp.Value;
+
+            foreach (var edge in edges)
+            {
+                string targetStepName = stepNames[edge.OutputTarget.StepId];
+
+                // Link edges without a specific preceding step to the Start node
+                if (!process.Steps.Any(s => s.Edges.ContainsKey(stepId)))
+                {
+                    sb.AppendLine($"{indentation}Start[Start] --> {targetStepName}[{targetStepName}]");
+                }
+            }
+        }
+
+        // Process each step
+        foreach (var step in process.Steps)
+        {
+            var stepId = step.State.Id;
+            var stepName = step.State.Name;
+
+            // Handle edges from this step
+            if (step.Edges != null)
```

Comment:
Does this work for subprocesses? As an example take a look at this test: https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/Experimental/Process.IntegrationTests.Shared/ProcessTests.cs#L87

Reply:
No, it will only draw the first level.
Maybe a next version to work in could be to enable X level of nesting?

Reply:
Now it does :)